lte/gateway/python/integ_tests/s1aptests/s1ap_utils.py

```diff
@@ -1005,6 +1005,7 @@ def restart_all_services(self):
                 "cd /home/vagrant/magma/lte/gateway/docker "
                 "&& docker-compose restart",
             )
+            self.wait_for_restart_to_finish(wait_time=30)
 
         self._wait_for_pipelined_to_initialize()
```

Comment:
For` docker`, the function `wait_for_restart_to_finish` is sufficient to check the status of pipelined also, so  `self._wait_for_pipelined_to_initialize()` will be unnecessary double verification for docker. Can we move this `self._wait_for_pipelined_to_initialize()` for `systemd` only between line 1002 and 1003.

Reply:
We discussed with the authors offline and deemed it a good idea to also have the check for a running datapath on openVSwitch since we have also seen tests failing with this scenario using a containerized AGW.
In order to minimize the waiting time, the initial sleep before checking for datapath was removed and the check-loop was adapted. 
Ping @nstng just FYI in case you want to request changes.

Reply:
Makes sense